oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderJwkSupport.java

```diff
@@ -135,8 +138,18 @@ private Jwt createJwt(String token, JWT parsedJwt) {
 
 			jwt = new Jwt(token, issuedAt, expiresAt, headers, jwtClaimsSet.getClaims());
 
+		} catch (RemoteKeySourceException ex) {
```

Comment:
I think we should leave it as is. 

We don't want to require services to be started in a specific order which would be the case if we checked at startup time. Just because it worked at startup doesn't mean we won't get the error later either.

It is quite possible they keys that are already loaded will work for many requests, so failing immediately for this request is quite reasonable as long as a previous key that was already loaded continues to work.